src/metrics/print.rs

```diff
@@ -1,23 +1,95 @@
-use super::Consumer;
+use std::io::Write;
+use std::sync::{Arc, Mutex};
+
 use async_trait::async_trait;
 use tokio::sync::mpsc::Receiver;
 
-pub struct Print;
+use super::Consumer;
+
+pub struct Print<T: 'static + Write + Send> {
+    csv_writer: Arc<Mutex<csv::Writer<T>>>,
+}
+
+impl<T: 'static + Write + Send> Print<T> {
+    pub fn new(writer: T) -> Self {
+        Self {
+            csv_writer: Arc::new(Mutex::new(csv::Writer::from_writer(writer))),
```

Comment:
since the `csv::Writer` is not `Send+Sync`, and I need to use the same instance of `csv::Writer::from_writer(writer)` every time I call `write_record_not_blocking `->`spawn_blocking `, I had to use the `Arc<Mutex<..>>`. Are there better solutions?

Reply:
With this configuration, testing code is difficult because:
- `csv::Writer` is behind a Mutex and this means I can only get a reference to it
- `csv::Writer` exposes the method `into_inner(self)` to get the inner `writer:T` that requires ownership of `csv::Writer`

Reply:
Hmm, that is a pain. I'm a bit surprised it is not `Send + Sync`. I'll try to check out this branch and play around a bit later to give better feedback.

Reply:
thanks, I realized I didn't write correctly (my bad):
the reason why I need to use the `Mutex` is that the `csv::Writer` requires a `&mut self` to write a record

Reply:
I pulled this down locally to explore yesterday and had quite a hard time finding a better way to do this 😆 

Unfortunately, nothing fruitful came from my adventure.

Reply:
thanks for your attempt!